feat: new clusters will use new strategy for enriching service attributes (#3428)
kyma-project/telemetry-manager: docs/user/filter-and-process/automatic-data-enrichment.md

>
> However, if you choose to use the OTel enrichment strategy, be aware of [these OTel-specific edge case limitations](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/k8sattributesprocessor/README.md#configuring-recommended-resource-attributes).

> [!WARNING]

> > The legacy strategy for service name enrichment will be deprecated. To avoid disruptions, migrate to the new OTel-based enrichment strategy by adding the annotation `telemetry.kyma-project.io/service-enrichment: otel` to your Telemetry CR.

Reviewer: I'm assuming that it will already be deprecated when we release 1.63.
Working on the deprecation note.

```suggestion
> The legacy strategy for service name enrichment is deprecated. To avoid disruptions, migrate to the new OTel-based enrichment strategy by adding the annotation `telemetry.kyma-project.io/service-enrichment: otel` to your Telemetry CR.
```
Instead of spelling out how to migrate here, can we link to the document that describes the new enrichment strategy?

Author: @NHingerl It won't be considered deprecated in release 1.63. We will deprecate it in the future with phase 3 here https://github.com/kyma-project/telemetry-manager/issues/2165. Probably actual deprecation will be in release 1.64. We should want to give an early warning and give users more time to switch to the new strategy

Reviewer: "Deprecation Note" is the warning. 
"Deletion Note" is what we use to announce we'll remove the feature in the future.

Author: As agreed, I changed the warning now to `is deprecated` instead of `will be deprecated`.

Author: The description of the new enrichment strategy is in the same document
